app/services/CartServiceImpl.java

```diff
@@ -145,11 +174,19 @@ private ProductVariant getVariantInContract(final ProductProjection product, fin
     }
 
     private Optional<ProductVariant> variant(final ProductProjection product, final String pactasId) {
-        final Optional<ProductVariant> variant = product.getAllVariants().stream()
-                .filter(v -> v.getAttribute(ShopKeys.ID_MONTHLY).getValue(AttributeAccess.ofString()).equals(pactasId))
-                .filter(v -> v.getAttribute(ShopKeys.ID_TWO_WEEKS).getValue(AttributeAccess.ofString()).equals(pactasId))
-                .filter(v -> v.getAttribute(ShopKeys.ID_WEEKLY).getValue(AttributeAccess.ofString()).equals(pactasId))
-                .findFirst();
-        return variant;
+        return product.getAllVariants().stream().map(variant -> {
+                    final Optional<String> monthly = Optional.of(variant.getAttribute(PactasKeys.ID_MONTHLY))
+                            .map((attribute) -> attribute.getValue(AttributeAccess.ofString()));
+                    final Optional<String> twoWeeks = Optional.of(variant.getAttribute(PactasKeys.ID_TWO_WEEKS))
+                            .map((attribute) -> attribute.getValue(AttributeAccess.ofString()));
+                    final Optional<String> weekly = Optional.of(variant.getAttribute(PactasKeys.ID_WEEKLY))
+                            .map((attribute) -> attribute.getValue(AttributeAccess.ofString()));
+
+                    if (pactasId.equals(monthly) || pactasId.equals(twoWeeks) || pactasId.equals(weekly)) {
```

Comment:
`pactasId` is of type String so equals with an Optional like `monthly` will always be false.